src/WorkOS.net/Services/DirectorySync/DirectorySyncService.cs

```diff
@@ -139,6 +139,25 @@ public async Task<User> GetUserAsync(string id, CancellationToken cancellationTo
             return await this.Client.MakeAPIRequestAsync<User>(request, cancellationToken);
         }
 
+        /// <summary>
+        /// Asynchronously deletes a Directory.
+        /// </summary>
+        /// <param name="id">Directory unique identifier.</param>
+        /// <param name="cancellationToken">
+        /// An optional token to cancel the request.
+        /// </param>
+        /// <returns>A Task wrapping a deleted WorkOS Directory record.</returns>
+        public async Task<Directory> DeleteDirectory(string id, CancellationToken cancellationToken = default)
```

Comment:
This return type isn't correct, as we don't return a `Directory` from the endpoint.

We just return a HTTP 202 with no body, so the return type for this method should just be `Task`.

Reply:
@maxdeviant When using `public async Task DeleteDirectory`, I seem to get this error:
```
Since 'DirectorySyncService.DeleteDirectory(string, CancellationToken)' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'?
```
Are there other adjustments needed to make this change?

Reply:
@logangingerich You can't return anything from a method that returns `Task` instead of `Task<T>`, so you'll need to discard the result of the API request:

```cs
await this.Client.MakeAPIRequestAsync<Directory>(request, cancellationToken)
```

instead of

```cs
return await this.Client.MakeAPIRequestAsync<Directory>(request, cancellationToken)
```